test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -674,6 +674,12 @@ def test_quantum_successors(self):
             or (isinstance(successor2, DAGOutNode) and isinstance(successor1.op, Reset))
         )
 
+        if isinstance(successor1, DAGOutNode):
+            self.assertEqual(repr(successor1), f"DAGOutNode(wire = {successor1.wire})")
+
+        if isinstance(successor2, DAGOutNode):
+            self.assertEqual(repr(successor2), f"DAGOutNode(wire = {successor2.wire})")
+
```

Comment:
It would be better to put these tests somewhere that only tests the `repr` method (like with the `DAGOpNode.__repr__` tests) - it's good to be logically grouped, but also we want a test failure to come with a descriptive name.  Here, if the test failed, it'd appear in the logs that `test_quantum_successors` failed, which is really unrelated, which makes things harder to track down easily (what the test suite is meant to help with).